packages/lesswrong/lib/collections/petrovDayActions/schema.ts

```diff
@@ -1,12 +1,33 @@
 import { userOwns } from "@/lib/vulcan-users/permissions";
+import { ALLOWABLE_COLLECTIONS } from "../moderationTemplates/schema";
+import { TupleSet, UnionOf } from "@/lib/utils/typeGuardUtils";
+
+const ACTION_TYPES = [
+  'optIn',
+  'hasRole',
+  'nukeTheWest',
+  'nukeTheEast',
+  'eastPetrovAllClear',
+  'eastPetrovNukesIncoming',
+  'westPetrovAllClear',
+  'westPetrovNukesIncoming'
+] as const
+
+const ACTION_TYPES_SET = new TupleSet(ACTION_TYPES)
+export type PetrovDayActionType = UnionOf<typeof ACTION_TYPES_SET>
+
 
 const schema: SchemaType<"PetrovDayActions"> = {
   // NOTE: this whole schema is bad, sorry Robert
   actionType: {
     type: String,
     nullable: false,
     canRead: ['guests'],
-    canCreate: ['members']
+    canCreate: ['members'],
+    allowedValues: [...ACTION_TYPES],
```

Comment:
you want to run `yarn generate` after adding this, since otherwise you don't get it into the generated types